snyk/Tiltfile

```diff
@@ -1,11 +1,41 @@
-def snyk(path, name='snyk'):
+
+def snyk(path, test_type, name='snyk', k8s_dep='', *options):
     """
     Args:
-      path: path to file to test
-      name: a name for the resource, defaults to snyk. Useful if you want multiple invocations
+      path: path to the file or container image (name:tag) to test
+      test_type: one of 'oss', 'container', or 'iac' corresponding to the various Snyk CLI tests
+      name: a name for the resource. useful for labeling when running multiple tests. defaults to 'snyk'
+      k8s_dep: used for container tests - sets the re-test dependency for Tilt. Ignored by other test types.
+      *options: additional CLI options; appended to the snyk command. see `snyk --help` for full CLI options.
+          - container: this is where to insert the 'file=path/to/Dockerfile' option
     """
+    if test_type == 'oss':
+        snyk_cmd = " ".join([
+            "snyk test --all-projects",
+            path
+        ])
+        test_deps = [path]
+    elif test_type == 'container':
+        snyk_cmd = " ".join([
+            "snyk container test",
+            path
+        ])
+        test_deps = [k8s_dep]
+    elif test_type == 'iac':
+        snyk_cmd = " ".join([
+            "snyk iac test",
+            path
+        ])
+        test_deps = [path]
+    else:
+        print("Test type should be one of oss, container, or iac.")
```

Comment:
should this be a `fail` command? i don't think we want this to fall through

Reply:
Fixed.